python/pyiceberg/exceptions.py

```diff
@@ -82,3 +82,7 @@ class NoSuchPropertyException(Exception):
 
 class NotInstalledError(Exception):
     """When an optional dependency is not installed"""
+
+
+class ColumnNotFoundError(Exception):
+    """When selecting a column that doesn't exist"""
```

Comment:
In Java, we just throw IllegalArgumentException (equivalent to ValueError, I think) when something can't be found.

Reply:
I figured this would be a nice one to add since it is very user facing. We're throwing a lot of `ValueError`'s already 😱